testutils/sims/runner.go

```diff
@@ -103,46 +109,82 @@ func RunWithSeeds[T SimulationApp](
 		seed := seeds[i]
 		t.Run(fmt.Sprintf("seed: %d", seed), func(t *testing.T) {
 			t.Parallel()
-			// setup environment
-			tCfg := cfg.With(t, seed, fuzzSeed)
-			testInstance := NewSimulationAppInstance(t, tCfg, appFactory)
-			var runLogger log.Logger
-			if cli.FlagVerboseValue {
-				runLogger = log.NewTestLogger(t)
-			} else {
-				runLogger = log.NewTestLoggerInfo(t)
-			}
-			runLogger = runLogger.With("seed", tCfg.Seed)
-
-			app := testInstance.App
-			stateFactory := setupStateFactory(app)
-			simParams, err := simulation.SimulateFromSeedX(
-				t,
-				runLogger,
-				WriteToDebugLog(runLogger),
-				app.GetBaseApp(),
-				stateFactory.AppStateFn,
-				simtypes.RandomAccounts, // Replace with own random account function if using keys other than secp256k1
-				simtestutil.SimulationOperations(app, stateFactory.Codec, tCfg, testInstance.App.TxConfig()),
-				stateFactory.BlockedAddr,
-				tCfg,
-				stateFactory.Codec,
-				app.TxConfig().SigningContext().AddressCodec(),
-				testInstance.ExecLogWriter,
-			)
-			require.NoError(t, err)
-			err = simtestutil.CheckExportSimulation(app, tCfg, simParams)
-			require.NoError(t, err)
-			if tCfg.Commit {
-				simtestutil.PrintStats(testInstance.DB)
-			}
-			for _, step := range postRunActions {
-				step(t, testInstance)
-			}
+			RunWithSeed(t, cfg, appFactory, setupStateFactory, seed, fuzzSeed, postRunActions...)
 		})
 	}
 }
 
+func RunWithSeed[T SimulationApp](
+	tb testing.TB,
+	cfg simtypes.Config,
+	appFactory func(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest bool, appOpts servertypes.AppOptions, baseAppOptions ...func(*baseapp.BaseApp)) T,
+	setupStateFactory func(app T) SimStateFactory,
+	seed int64,
+	fuzzSeed []byte,
+	postRunActions ...func(t testing.TB, app TestInstance[T]),
+) {
+	tb.Helper()
+	// setup environment
+	tCfg := cfg.With(tb, seed, fuzzSeed)
+	testInstance := NewSimulationAppInstance(tb, tCfg, appFactory)
+	var runLogger tlog.Logger
+	if cli.FlagVerboseValue {
+		runLogger = tlog.NewTestLogger(tb)
+	} else {
+		runLogger = tlog.NewTestLoggerInfo(tb)
+	}
+	runLogger = runLogger.With("seed", tCfg.Seed)
+
+	app := testInstance.App
+	stateFactory := setupStateFactory(app)
+	ops, reporter := prepareWeightedOps(app.SimulationManager(), stateFactory, tCfg, testInstance.App.TxConfig(), runLogger)
+	simParams, err := simulation.SimulateFromSeedX(tb, runLogger, WriteToDebugLog(runLogger), app.GetBaseApp(), stateFactory.AppStateFn, simtypes.RandomAccounts, ops, stateFactory.BlockedAddr, tCfg, stateFactory.Codec, testInstance.ExecLogWriter)
+	require.NoError(tb, err)
+	err = simtestutil.CheckExportSimulation(app, tCfg, simParams)
+	require.NoError(tb, err)
+	if tCfg.Commit {
+		simtestutil.PrintStats(testInstance.DB, tb.Log)
+	}
+	// not using tb.Log to always print the summary
+	fmt.Printf("+++ DONE (seed: %d): \n%s\n", seed, reporter.Summary().String())
+	for _, step := range postRunActions {
+		step(tb, testInstance)
+	}
+}
+
+type (
```

Comment:
I think simulations interfaces should maybe get a place in core/appmodule(v2).
`type/module` is actually for legacy stuff, so the legacy interfaces can stay there but it would be cool if the new one could be in core, as it is the central place of module APIs.